test/run_envoy_fuzz_coverage.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+
+set -e
+
+[[ -z "${SRCDIR}" ]] && SRCDIR="${PWD}"
+
+echo "Starting run_envoy_fuzz_coverage.sh..."
+echo "    PWD=$(pwd)"
+echo "    SRCDIR=${SRCDIR}"
+
+# This is the fuzz target that will be run to generate coverage data.
+if [[ $# -gt 0 ]]; then
+  FUZZ_TARGETS=$*
+else
+  # By default, this will be all fuzz targets.
+  FUZZER_TARGETS_CC=$(find . -name *_fuzz_test.cc)
+  FUZZER_TARGETS="$(for t in ${FUZZER_TARGETS_CC}; do echo "${t:2:-3}"; done)"
+  FUZZ_TARGETS=""
+  for t in ${FUZZER_TARGETS}
+  do
+    FUZZ_TARGETS+="//"$(dirname "$t")":"$(basename "$t")" "
+  done
+fi
+
+echo ${FUZZ_TARGETS}
+
+# Build all fuzz targets to run instrumented with libfuzzer in sequence.
+echo "Building fuzz targets..."
+for t in ${FUZZ_TARGETS}
+do
+  bazel build "${t}_with_libfuzzer" --config asan-fuzzer -c opt
+done
+
+# Now run each fuzz target in parallel for 60 seconds.
+pids=""
+TEMP_CORPORA=""
+echo "Running fuzz targets..."
+for t in ${FUZZ_TARGETS}
+do
+  # Get the original corpus for the fuzz target
+  ORIGINAL_CORPUS=$(bazel query "labels(srcs, ${t}_corpus)" | head -1)
+  ORIGINAL_CORPUS=${ORIGINAL_CORPUS/://}
+  ORIGINAL_CORPUS=$(dirname ${ORIGINAL_CORPUS})
+  # Create temp directory in target's corpus
+  CORPUS_DIR=$(mktemp -d -p $(pwd)/${ORIGINAL_CORPUS:2})
+  TEMP_CORPORA+="${CORPUS_DIR} "
+  # Run fuzzing process.
+  TARGET_BINARY="${t/://}"
+  bazel-bin/${TARGET_BINARY:2}_with_libfuzzer -max_total_time=60 ${CORPUS_DIR} $(pwd)${ORIGINAL_CORPUS:1} &
```

Comment:
Does this need to be a subdirectory of the original corpus? I'd naively `mkdtemp` in `TMPDIR` to simplify.

Reply:
Ignore this comment, I wrote this because I didn't have the high-level "fuzz, create corpus, do coverage run" flow in mind. Might be worth some top-level comments.